### PR TITLE
[AutoDiff] Remove primal/adjoint from user syntax and SIL.

### PR DESCRIPTION
**Pull request by rxwei**
This is the final step in switching to JVP/VJP.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

File: include/swift/AST/Attr.h [hunk 1]
```diff
@@ -1302,18 +1302,14 @@ class ClangImporterSynthesizedTypeAttr : public DeclAttribute {
...
-/// until we have fully switched to 'jvp' and 'vjp'.
+/// custom associated autodiff functions: 'jvp' and 'vjp'.
 ///
 /// Note: 'jvp' and 'vjp' are not fully supported yet. In particular, the core
 /// AD pass does not use them. We are incrementally adding support for them.
 ///
 /// For example:
-///   @differentiable(reverse, adjoint: foo(_:_:seed:) where T : FloatingPoint)
-///   @differentiable(reverse, wrt: (self, .0, .1), adjoint: bar(_:_:_:seed:))
+///   @differentiable(reverse, jvp: jvpFoo where T : FloatingPoint)
```
Comment (dan-zheng) [inline include/swift/AST/Attr.h]: Remove `reverse`, on the following line too.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow Linux

File: include/swift/SIL/SILFunction.h [hunk 2]
```diff
@@ -103,15 +103,6 @@ class SILSpecializeAttr final {
 /// attribute is used by the automatic differentiation pass to find the autodiff
 /// functions associated with a function: 'primal', 'adjoint', 'jvp', and 'vjp'.
```
Comment (dan-zheng) [inline include/swift/SIL/SILFunction.h]: Remove mentions of `primal` and `adjoint`

File: lib/SIL/SILFunctionBuilder.cpp [hunk 3]
```diff
@@ -76,31 +76,11 @@ SILFunctionBuilder::addFunctionAttributes(SILFunction *F, SILDeclRef constant,
       // Either only adjoint is specified, or both primal and adjoint are
```
Comment (dan-zheng) [inline lib/SIL/SILFunctionBuilder.cpp]: Remove mentions of `primal` and `adjoint`, remove `primName` and `adjName` below.
Also, `spcified` -> `specified`.

File: test/AutoDiff/differentiable_attr_access_control.swift [hunk 4]
```diff
@@ -4,29 +4,22 @@
 // its primal/adjoint must also be exported.
```
Comment (dan-zheng) [inline test/AutoDiff/differentiable_attr_access_control.swift]:
```suggestion
 // its JVP/VJP must also be exported.
```

File: test/AutoDiff/method.swift [hunk 5]
```diff
@@ -171,94 +171,80 @@ extension Float {
...
   }
 
-  static func dSquared(_ seed: Float, _ origVal: Float, _ p: CustomParameter)
-      -> CustomParameter {
-     return CustomParameter(x: (2 * p.x).clamped(to: -10.0...10.0) * seed)
+  static func dSquared(_ p: CustomParameter) -> (Float, (Float) -> CustomParameter) {
+    return (p.x * p.x, { v in CustomParameter(x: (2 * p.x).clamped(to: -10.0...10.0) * v) })
   }
 
   // There is currently no way to define multiple custom adjoints wrt different
```
Comment (dan-zheng) [inline test/AutoDiff/method.swift]: Rename `adjoint` to `VJP`

File: test/AutoDiff/superset_adjoint.swift [hunk 6]
```diff
@@ -5,17 +5,16 @@ import StdlibUnittest
 
 var SupersetAdjointTests = TestSuite("SupersetAdjoint")
```
Comment (dan-zheng) [inline test/AutoDiff/superset_adjoint.swift]:
```suggestion
var SupersetVJPTests = TestSuite("SupersetVJP")
```

**Review by dan-zheng — APPROVED** [review]
This is big! 🎉
(Feel free to ignore comments, remaining mentions of `primal` and `adjoint` can be cleaned up later.)

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Pull request merged**